src/rfactor/process.py

```diff
@@ -1,11 +1,110 @@
 import re
+from functools import wraps
 from pathlib import Path
 
 import numpy as np
 import pandas as pd
+from pandas import Timedelta
 from tqdm import tqdm
 
 
+def valid_column(rain, req_col):
```

Comment:
I would move the validation functionalities to a separate file/module as these are not part of the processing itself. For example `validation.py`, `check.py`,...

Reply:
Done